src/Functions/parseDateTime.cpp

```diff
@@ -0,0 +1,1747 @@
+                                instructions.emplace_back(ACTION_ARGS_WITH_BIND(Action::jodaMonthOfYear, repetitions));
+                            else
+                                instructions.emplace_back(ACTION_ARGS_WITH_BIND(Action::jodaMonthOfYearText, repetitions));
+                            break;
+                        case 'd':
+                            instructions.emplace_back(ACTION_ARGS_WITH_BIND(Action::jodaDayOfMonth, repetitions));
+                            break;
+                        case 'a':
+                            instructions.emplace_back(ACTION_ARGS_WITH_BIND(Action::jodaHalfDayOfDay, repetitions));
+                            break;
+                        case 'K':
+                            instructions.emplace_back(ACTION_ARGS_WITH_BIND(Action::jodaHourOfHalfDay, repetitions));
+                            break;
+                        case 'h':
+                            instructions.emplace_back(ACTION_ARGS_WITH_BIND(Action::jodaClockHourOfHalfDay, repetitions));
+                            break;
+                        case 'H':
+                            instructions.emplace_back(ACTION_ARGS_WITH_BIND(Action::jodaHourOfDay, repetitions));
+                            break;
+                        case 'k':
+                            instructions.emplace_back(ACTION_ARGS_WITH_BIND(Action::jodaClockHourOfDay, repetitions));
+                            break;
+                        case 'm':
+                            instructions.emplace_back(ACTION_ARGS_WITH_BIND(Action::jodaMinuteOfHour, repetitions));
+                            break;
+                        case 's':
+                            instructions.emplace_back(ACTION_ARGS_WITH_BIND(Action::jodaSecondOfMinute, repetitions));
+                            break;
+                        case 'S':
+                            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "format is not supported for fractional seconds");
+                            break;
+                        case 'z':
+                            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "format is not supported for timezone");
+                            break;
+                        case 'Z':
+                            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "format is not supported for timezone offset id");
+                        default:
+                            if (isalpha(*cur_token))
+                                throw Exception(
+                                    ErrorCodes::NOT_IMPLEMENTED, "format is not supported for {}", String(cur_token, repetitions));
+
+                            instructions.emplace_back(String(cur_token, pos - cur_token));
+                            break;
+                    }
+                }
+            }
+#undef ACTION_ARGS_WITH_BIND
+        }
+
+
+        ALWAYS_INLINE String getFormat(const ColumnsWithTypeAndName & arguments) const
```

Comment:
Minor: It would be nice to move utility functions `getFormat` and `getTimeZone` to the top of the file to have all helpers in one place.

Reply:
There is just a single caller of `getFormat`, perhaps we don't need a separate utility function.

Reply:
Both getFormat and getTimeZone reply on IFunction::getName(), it is impossible to move them to top of the file without modifying current implementation.